ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -385,24 +391,43 @@ void onPutBlobError(Exception e) {
    * @param options the {@link GetBlobOptions} associated with the request.
    */
   void onGetBlobError(Exception e, GetBlobOptions options) {
+    GetOperationType type =
+        options != null && options.getGetOperationType() != null ? options.getGetOperationType() : GetOperationType.All;
+    switch (type) {
+      case All:
+      case Data:
+        onGetBlobDataError(e, options);
+        break;
+      case BlobInfo:
+        onGetBlobInfoError(e);
+        break;
+    }
```

Comment:
I think you can guarantee that `options.getOperationType` is non-null in the `GetBlobOptions` constructor - unlike the range, an operation always has an operation type. So you could avoid the second null check above.

This could be made simpler I think:
```java
if (options != null && options.getOperationType() == GetBlobOptions.OperationType.BlobInfo) {
  onGetBlobInfoError(e);
} else {
  onGetBlobDataError(e, options);
}
```